### PR TITLE
Fix: Rocket Buggy Missile Deleted Before Reaching Reinforcement Pad At Max Distance

### LINKED ISSUE #142: GLA Rocket Buggy Can't Damage Reinforcement Pad From Maximum Distance
Buggy Rockets Seem to Fall off before reaching their target if it was A Reinforcement Pad

+ Moving A bit Closer will get it to Damage it.
+ Force Firing on the Edges will naturally make buggy move Closer therefore Damage the Target.
+ Problem is on All maps, Enemy, Player, or Netrual Pad, Force firing and Regular Targetting all give the same Result.
+ Tested other Tech Buildings Repair Pad, Artillery, Oil, Refinery, and all Get Damaged normally.

https://user-images.githubusercontent.com/83613147/184507981-00fecdc8-1ad6-4349-9b33-9be1dcad70cb.mp4

### PR DESCRIPTION
**Pull request by commy2**
- fix https://github.com/TheSuperHackers/GeneralsGameCode/issues/142
- close https://github.com/TheSuperHackers/GeneralsGamePatch/pull/931

1850 (+1 frame) wasn't enough to make all rockets hit, only half of them did.

### CODE REVIEW

**Comment by xezon** [comment]
What would happen if the fuel time was 9999 ?

File: Patch104pZH/GameFilesEdited/Data/INI/Object/WeaponObjects.ini [hunk 1]
```diff
@@ -3019,7 +3019,7 @@ Object RocketBuggyMissile
   End
   Behavior = MissileAIUpdate ModuleTag_07
     TryToFollowTarget               = Yes
-    FuelLifetime                    = 1800
+    FuelLifetime                    = 1900 ; Patch104p @bugfix commy2 19/08/2022 Give just enough range to hit Reinforcement Pads at max range. Was 1800.
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/WeaponObjects.ini:3022]: What would happen if the fuel time was 9999 ?

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/WeaponObjects.ini:3022]: The missile would follow moving targets for roughly 10 seconds before reaching them or running out of fuel and being deleted, making it effectively impossible to dodge them by making your units retreat.

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/WeaponObjects.ini:3022]: Ok. So the 100 ms added here are a tiny buff in principle, as it allows Buggies to hit evading units for 5.5% longer time, correct?

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/WeaponObjects.ini:3022]: Pretty much.

**Comment by MTKing4** [comment]
This is a buff for buggies isn't it?

I thought that we needed to adjust reinforcement pad itself not the buggy, something about its bounding box being too big or something? Because buggy doesn't have this issue with any other building, surely the problem is with the building not the unit.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/WeaponObjects.ini (hunk 1 above)
Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/WeaponObjects.ini:3022]: Ok. In that case the other approach likely is preferable.

**Comment by commy2** [comment]
It's the only building that is flat in its center, aside from the Supply Drop Zone, but that one is not as wide.
Range of the Buggy has to be buffed, or the missile has to explode when running out of fuel. No other work around, aside from remodelling the Reinforcement Pad.

**Comment by MTKing4** [comment]
Well can't you make it so that missiles just collide with the geometry of the building? Meaning they do it on the edge of the model itself, if modelling is the only other solution then I prefer that over doing changes on the buggy

**Comment by commy2** [comment]
> Well can't you make it so

Idk. I won't make that anytime soon.

**Pull request closed without merging**